File: Electronics_Shop_17.Model/Requests/UpdateOrder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Model.Requests
{
    public class UpdateOrder
    {
        public decimal? TotalAmount { get; set; }

        public int? AdressId { get; set; }
        public int? CouponId { get; set; }
        public int? PaymentMethodId { get; set; }
        public virtual List<UpdateOrderItem>? OrderItems { get; set; }
    }
}

File: angular17_web/src/app/models/order/update.order.ts
import { UpdateOrderItem } from '../order-item/update.order.item';

export class UpdateOrder {
  totalAmount?: number;
  adressId?: number;
  couponId?: number;
  paymentMethodId?: number;
  orderItems?: UpdateOrderItem[];

  constructor(data?: any) {
    if (!data) return;

    this.totalAmount = data.totalAmount;
    this.adressId = data.adressId;
    this.couponId = data.couponId;
    this.paymentMethodId = data.paymentMethodId;
    this.orderItems = Array.isArray(data.orderItems)
      ? data.orderItems.map((x: any) => new UpdateOrderItem(x))
      : [];
  }
}
